.github/actions/check-argocd-status/action.yml

```
@@ -0,0 +1,112 @@
+                REASON=$(echo "$MANIFEST" | jq -r '.status.conditions[] | select(.type=="Progressing") | .reason')
+                if [[ "$IMAGE_TAG" == "$TARGET_IMAGE_TAG" && "$REASON" == "NewReplicaSetAvailable" && "$STATUS" == "True" ]]; then
+                    echo "✅ Deployment rollout successful"
+                    break
+                else
+                    echo "⏳ Resource is still running... Waiting for completion."
+                fi
+            fi
+
+            # Wait before the next check
+            sleep 10
+        done
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)